crates/storage/provider/src/providers/mod.rs

```diff
@@ -99,8 +103,47 @@ impl<DB: Database> BlockProvider for ShareableDatabase<DB> {
         Ok(ChainInfo { best_hash, best_number, last_finalized: None, safe_finalized: None })
     }
 
-    fn block(&self, _id: BlockId) -> Result<Option<Block>> {
-        // TODO
+    fn block(&self, id: BlockId) -> Result<Option<Block>> {
+        let block_number = match id {
+            BlockId::Number(num) => self.convert_block_number(num)?,
+            BlockId::Hash(hash) => self.block_number(H256(hash.0))?,
+        };
```

Comment:
I feel like we need another convert function for BlockId?
def useful for more function I think

Reply:
ah, we already have this function, i just missed it. fixing...